### ISSUE TITLE
Pure ProxyNCA experiment

### ISSUE THREAD

**Issue by alpoler** [euwern/proxynca_pp#6]
I change loss function to ProxyNCA to make experiment of it but I observe that loss function value comes negative. I feel suspicious about accuracy of my configuration although i use parameters from the original paper. Could you share your experiment setup for only ProxyNCA ?

**Comment by phoenixinwater** [comment]
Hi alpoler, I got negative loss values as well. I found most other implementations will use softmax which appears to mitigate the problem at the expense of not being true to the original paper.

**Comment by euwern** [comment]
Original proxynca [1] doesn't have softmax (algorithm 1). There is no guarantee that k = (exp(-positive_distance)/ exp(-negative_distance)) is below 1. if k is greater than one, algorithm 1 will give you negative loss values. (-log(k))

You will most likely see negative loss values when you set the temperature scale to 3, where it amplifies k. In the original proxynca, there is no temperature scale (scale=1). I suspect the negative loss still happens, but it happens less frequently and, on average, gives you positive loss values. I have attached the config to replicate the original proxynca below. 

[1]https://openaccess.thecvf.com/content_ICCV_2017/papers/Movshovitz-Attias_No_Fuss_Distance_ICCV_2017_paper.pdf

	{
		"name": "str('example')",
		"gpu_id": "int(0)",
		"criterion": {
			"type": "loss.ProxyNCA_classic",
			"args": {
				"scale": "1"
			}
		},
		"lr_scheduler": {
			"type": "torch.optim.lr_scheduler.ReduceLROnPlateau",
			"args": {
				"mode": "str('max')",
				"patience": "4*1"
			}
		},
		"lr_scheduler2": {
			"type": "torch.optim.lr_scheduler.MultiStepLR"
			
		},
	   
		"nb_epochs": "40*1",
		"sz_batch": "32*1",
		"sz_embedding": "2048",
		"num_class_per_batch": "32",
		"num_gradcum": "1",
		"is_random_sampler":"True",
		"transform_key": "str('transform_parameters')",
		"model":{
			"type": "networks.Feat_resnet50_avg"
		},
		"opt": {
			"type": "torch.optim.Adam",
			"args": {
				"embedding": {
					"weight_decay": "0"
				},
				"backbone": {
					"weight_decay": "0"
				},
				"proxynca": {
					"weight_decay": "0" 
				},
				"base": {
					"lr": "4e-3",
					"eps": "1.0"
				}
			}
		}
	}
